src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,264 @@
+
+5. In a new terminal window, repeat the commands `cd /workspaces/WRoverPlayground`, `catkin_make -DCMAKE_EXPORT_COMPILE_COMMANDS=ON`, and `source devel/setup.bash` in this order. 
+
+6. With everything hopefully set up properly, you may now run `rosrun topic_training Demo` to execute the C++ version of the demo, or `rosrun topic_training Demo.py` to execute the Python version of the demo. 
+In the terminal, you should see the program continually print "currently at {next sine value}" until you exit the program by inputting `Ctrl+C` in the terminal or the program crashes.
+
+### rqt_graph and rqt_plot
+1. In another terminal window, run `rosrun rqt_graph rqt_graph`. 
+You should already be familiar with this command from the ROS Debug/Tools Training. 
+
+2. Make sure the graph representation is what is expected. 
+It should show "/PlotDemo" enclosed in a circle pointing to "/Demo" with an arrow named "/plot".
+
+You should have also learned about rqt_plot. 
+
+1. Type Ctrl-C into the terminal to stop running Demo(.cpp or .py) so only PlotDemo (.cpp or .py) is running. 
+
+2. Run `rosrun rqt_plot rqt_plot`. 
+You should see a sine graph being plotted onto the screen. Recall that rqt_plot can only be used with numerical values. Using it with the file Demo(.cpp or .py) would not have worked because that is dealing with Strings. Feel free to check this yourself.
+
+3. You may also look through the file PlotDemo(.cpp or .py) to see how the math works and to see another example of a Publisher but with a different message type. 
+For simplicity's sake, the comments in these files are only explaining the math since you should already understand the basic ROS functions from the previous section of this training.
+
+While developing your code for the challenge, use rqt_graph and rqt_plot to make sure your nodes are working as expected.
+
+## Defining a Custom Topic Type
+A topic's type is determined by the type of the message being published. If you run the command `rostopic type chatter`, you should see the terminal print out "std_msgs/String" while roscore and rosrun are executing Demo(.cpp or .py). The command will print out "std_msgs/Float64" if you are executing PlotDemo(.cpp or .py).
+
+A topic can have messages of type int8, int16, int32, int64, uint*, float32, float64, String, time, duration, other msg files, and array[] of variable length or fixed length. To message a different type, just change where ever there would be a String in Demo(.cpp or .py) to your desired type and update the includes (C++) or imports (Python) at the top of the file. Remember though, use message types instead of variable types, that is, use std_msgs/String (for C++) or std_msgs.msg.String (for Python) instead of just String.
+
+Along with the types listed above, you can also create custom topic types. Custom topic types are defined in .msg files. 
+
+1. In /workspaces/WRoverPlayground/src/topic-training/msg/, find the CustomType.msg file. 
```

Comment:
Same as before, adding 'code' to make the file path more readable. 

```suggestion
1. In `/workspaces/WRoverPlayground/src/topic-training/msg/`, find the CustomType.msg file. 
```

Reply:
Fixed in 5182f6b.